### ISSUE TITLE
User Guide Problems

### ISSUE THREAD

**Issue by DarioS** [suhrig/arriba#90]
It has `-k /path/to/known_fusions.tsv.gz -t /path/to/known_fusions.tsv.gz` in workflow.md but `-t` is for tags. Also, the user guide has an Edit on GitHub icon on the top-right of the reader's screen, but clicking on it causes a Error 404. Moreover, I searched the entire user guide for RefSeq_viral_genomes_v2.0.0.fa.gz and no matches were found. I don't know how to specify its path to arriba. Lastly, specifying `-p` results in `ERROR: Unknown option: -p`. Isn't it instead for draw_fusions.R rather than arriba?

**Comment by suhrig** [comment]
My apologies that the instructions are not detailed enough and cause so much confusion. I try to keep them short to spare users from having to read more information than they actually need to get started. I see how this can have the opposite effect for users who do not "blindly" copy and paste the commands from the user guide but make an effort to understand how they work.

To answer your questions:

> It has -k /path/to/known_fusions.tsv.gz -t /path/to/known_fusions.tsv.gz in workflow.md but -t is for tags.

This is intentional. It is true that `-t` is for tags, but you can pass the same file for both `-k` and `-t`. The file is formatted in a way that works for both parameters. `-k` only uses the first two columns, whereas `-t` uses all three. `-k` uses the file to apply sensitive filtering criteria and `-t` uses the file to tag fusions with the value from column three. The tags are annotated in Arriba's output file column `tags`. Currently, all tags are set to `Mitelman`, so this file essentially tags all recurrent fusions known in the Mitelman database. In theory, users could craft their own tags file with tags other than `Mitelman`. For example, users could create a file to tag fusions observed in normal tissue or potential artifacts or fusions with genes of interest.

I should probably clarify the use of the same file for both arguments in the workflow.md page.

> the user guide has an Edit on GitHub icon on the top-right of the reader's screen

The wiki is subject to Git version control. External users do not have write permission to Arriba's Git repository and therefore cannot make changes. I like to keep it this way, because I don't like the idea that anyone can edit the wiki. If you want to make an edit, you have to fork the Git repo, make the change, and submit a pull request. Alternatively, you can just send me an e-mail or open an issue like this one to notify me about mistakes. I am perfectly fine with getting notified this way, as I understand that the pull request method is cumbersome.

> I searched the entire user guide for RefSeq_viral_genomes_v2.0.0.fa.gz

It is not mentioned in the user guide, because it is not an input file of Arriba. This file is used to build a STAR index when a user seeks to detect viral integration sites. It is concatenated to the human genome. The file is mentioned in the script `download_references.sh`. When this script is called with an assembly name containing the keyword `viral`, it builds a STAR index from a concatenated human+viral assembly. Theoretically, I could download the viral genomes dynamically from the RefSeq website (just like it is done for the human genome), which would eliminate the need to store the file in the `database` directory and thus avoid confusion on the part of the user. But in contrast to the human genome, the RefSeq viral genomes are not versioned properly. You get different files depending on when you download. In a few months (or weeks) from now the download will be different. RefSeq even changes the file structure (!), which makes automated downloads difficult. So I have no choice but to store it in the `database` folder.

> Lastly, specifying -p results in ERROR: Unknown option: -p. Isn't it instead for draw_fusions.R rather than arriba?

Since version 2, Arriba can annotate protein domains itself (see column `retained_protein_domains`). This is useful if you want to screen the output files programmatically for the presence of protein domains, which is difficult to accomplish with the PDF output file format of `draw_fusions.R`.

Nevertheless, the error message you are seeing is a (minor) bug. It should read `ERROR: option -p requires an argument`. The option works, though. Just give it the protein domains file as an argument and all will be fine. I simply forgot to adapt the error message in case the user forgets to pass an argument. I will fix this for the next release. Thanks for reporting all of this and helping me polish the manual and code.

**Comment by DarioS** [comment]
Thanks for the clarification. I thought that Arriba might use the viral sequences directly like the DNA structural variants program I use named GRIDSS. It expects a standard human genome reference for alignment and annotates soft-clipped read alignments using Kraken 2, so it doesn't require existing data sets to be realigned to a combined reference genome. It will be nice to see that requirement explicitly mentioned. Protein domains works fine now that I deleted an old arriba from my PATH.

**Comment by suhrig** [comment]
> It expects a standard human genome reference for alignment and annotates soft-clipped read alignments 

I have considered this method many times, but ultimately opted for complete realignment for the following reasons:

- In contrast to DNA-Seq data (which GRIDSS was developed for) RNA-Seq data are small. So a complete realignment is not as costly as it is with WGS. It only takes about half an hour to do a complete realignment of a RNA-Seq sample.
- Only considering soft-clipped alignments ignores a good chunk of other informative reads, notably unmapped reads. For the best detection rate, these should also be screened for signs of viral infection. So the amount of reads that need to be realigned is considerably higher than just the soft-clipped reads, which further reduces the cost-benefit of taking existing alignments.
- Alignment algorithms often force reads to be mapped to human genome (with many mismatches) if they are not given the right reference genomes. That is, a viral read may be mapped to a human locus with similar sequence (without soft-clipping) and thus be invisible to the method that only considers soft-clipping. Though I have to admit I have no clue what the magnitude of this problem is, because I never investigated it.
- Aligning against concatenated genomes has the additional benefit that you can quantify expression of viral genomes. This is irrelevant to GRIDSS, because it deals with DNA-Seq data, but very helpful for detecting viral integration sites from RNA-Seq data. And the user can use this information for other types of analyses.
- There are other reasons to do complete realignment (at least, if you have alignments with STAR < 2.7.6a), because with alignments from the latest version, you get the additional benefit of improved chimeric read detection.
- Implementing this method is way more complex than complete realignment. I would be more complex to develop and more complex for end-users to install on their systems.

In summary, this method produces inferior results for more work and that is why I dismissed it. Maybe in the future I will provide scripts for this method to users who are fine with having results of a little bit lower quality in favor of computational efficiency. I did not consider Kraken2, however. Maybe that changes my mind if it turns out to be super easy with the help of this tool. I will have a look. As always, your input is very appreciated!

**Comment by DarioS** [comment]
You have convinced me that a combined reference for alignment is valuable!

**Comment by suhrig** [comment]
Hi Dario,

I recently pushed a script to the repo which allows Arriba to be run on existing alignments:

https://github.com/suhrig/arriba/blob/develop/scripts/run_arriba_on_prealigned_bam.sh

It still performs an alignment against concatenated human+viral genomes, but only on clipped or unmapped reads. All reads that have good mapping quality to the human genome as given in the prealigned BAM file are retained as they are. This means that you can use your standard RNA-Seq alignment workflow with a regular human genome (no viral sequences), and at the same time, you can use Arriba to detect viral integration sites, without the need to realign everything against concatenated genomes. In my tests, the script typically used only 10-15% of the CPU time of what a complete realignment would take.

The script has two limitations:

- As described above, the fusion calls will differ slightly from a complete realignment and may be inferior. Officially, I still recommend doing a complete realignment, because this method is well-tested.
- The script does not scale well. It does not parallelize beyond 2 CPUs, because the script is mostly concerned with extracting relevant reads from the prealigned BAM file, and this process is not easy to parallelize.

Regards,
Sebastian
